routes/utils.go

```diff
@@ -0,0 +1,25 @@
+package routes
+
+import (
+	"encoding/hex"
+	"github.com/deso-protocol/core/lib"
+	"github.com/pkg/errors"
+)
+
+// decodeBlockHashFromHex Decodes a BlockHash given a valid hex encoding. If the input does not represent a valid
+// BlockHash, this returns the corresponding error
+func decodeBlockHashFromHex(hexEncoding string) (*lib.BlockHash, error) {
```

Comment:
Moving forward, we should be using this function everywhere we decode block hashes from hex strings. I'll have another PR that migrates existing uses to use this function.